### PR TITLE
[wip] V2 api

### PR DESCRIPTION
**Pull request by davidenko87**
## Proposed changes

With the release of WeTransfer API V2 we need to update the SDK as well. We added new featutes to distinct transfer from boards. Where transfer are immutable and boards are mutable. Which means, after you create a board you can add files and web_items to it, but for transfers that isn't allowed. Also in this version we want to give the client the control of handling the upload. Previous version worked well as CLI but it couldn't handle large files when implemented on a server. 

## Types of changes

What types of changes does your code introduce to wetransfer_ruby_sdk?
_Put an `x` in the boxes that apply_

- [x] Docs (missing or updated docs)
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x]  Improvement / maintenance (removing technical debt, performance, tooling, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/WeTransfer/wetransfer_ruby_sdk/blob/master/.github/CONTRIBUTING.md) doc
- [x] I have added necessary documentation (if appropriate)
- [x] Code changes are covered by unit tests.
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/we_transfer_client/remote_board.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    [url, CHUNK_SIZE]
+  end
+
+  def prepare_file_completion(client:, file:)
+    file.complete_board_file(client: client, board_id: @id)
+  end
+
+  private
+
+  def to_instances(items:)
```
Comment (bermannoah) [inline lib/we_transfer_client/remote_board.rb]: what does this do?

**Comment by bermannoah** [comment]
Also I think we can remove 2.1 support (the core ruby team [no longer supports it](s/2017/04/01/support-of-ruby-2-1-has-ended/)). If we want to be extremely generous we can support 2.2, but that one's also been EOL'd this summer.

File: lib/we_transfer_client/remote_board.rb (hunk 1 above)
Comment (arnoFleming) [reply lib/we_transfer_client/remote_board.rb]: it instantiates all files or links into `RemoteFiles` or `RemoteLinks`

File: spec/we_transfer_client/remote_transfer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  describe '#Getters' do
+    subject { described_class.new(params) }
+
+    it '#files' do
+      subject.files
+    end
+
+    it '#url' do
+      subject.url
```
[the remark this answers is not in the document]
Comment (arnoFleming) [reply spec/we_transfer_client/remote_transfer_spec.rb]: it isn't tedious, but it does make sure it doesn't raise

File: spec/we_transfer_client_spec.rb [hunk 3]
```diff
@@ -2,8 +2,76 @@
 
 require_relative '../lib/we_transfer_client.rb'
 
-describe WeTransferClient do
+describe WeTransfer::Client do
+  subject { described_class.new(params) }
+  let(:params) { { api_key: ENV.fetch('WT_API_KEY') } }
```
[the remark this answers is not in the document]
Comment (arnoFleming) [reply spec/we_transfer_client_spec.rb]: It is used in the line above, in all instances of the class

File: wetransfer.gemspec [hunk 4]
```diff
@@ -5,7 +5,7 @@ require 'we_transfer_client/version'
 
 Gem::Specification.new do |spec|
   spec.name          = 'wetransfer'
-  spec.version       = WeTransferClient::VERSION
+  spec.version       = WeTransfer::VERSION
   spec.authors       = ['Noah Berman', 'David Bosveld']
```
[the remark this answers is not in the document]
Comment (arnoFleming) [reply wetransfer.gemspec]: fixed

File: wetransfer.gemspec [hunk 5]
```diff
@@ -5,7 +5,7 @@ require 'we_transfer_client/version'
...
   spec.email         = ['noah@wetransfer.com', 'david@wetransfer.com', 'developers@wetransfer.com']
```
[the remark this answers is not in the document]
Comment (arnoFleming) [reply wetransfer.gemspec]: IMHO: Nope, post-install warnings are frowned upon.

File: spec/we_transfer_client/remote_transfer_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (arnoFleming) [reply spec/we_transfer_client/remote_transfer_spec.rb]: or return false

**Pull request merged**